Firestore/core/src/firebase/firestore/immutable/llrb_node.h

```diff
@@ -273,6 +360,62 @@ void LlrbNode<K, V>::FixUp() {
   }
 }
 
+/**
+ * Fixes the root node so its color is always black.
+ *
+ * This change is safe because a red `root` must be a new root:
+ *   * If the key is not found, InnerErase returns an existing root, but
+ *     existing roots will already be black (by the invariant).
+ *   * If the key is found, InnerErase returns a new root, which is safe to
+ *     modify.
+ */
+template <typename K, typename V>
+void LlrbNode<K, V>::FixRootColor() {
+  if (red()) {
+    rep_->color_ = Color::Black;
+  }
+}
+
+template <typename K, typename V>
+void LlrbNode<K, V>::RemoveMin() {
+  // If the left node is empty then the right node must empty (because the tree
+  // is left-leaning) and this node must be the minimum.
+  if (left().empty()) {
+    rep_ = EmptyRep();
```

Comment:
Entirely optional: this could be `*this = LlrbNode{}`, which would be, IMO, a little more symmetrical with how `InnerErase` returns `LlrbNode{}`s.